### ISSUE TITLE
Relative path to source file in sourcemap is broken sometimes

### ISSUE THREAD

**Issue by glmars** [scala-js/scala-js#3135]
I found stable reproduction (see example project https://github.com/glmars/scalajs-wrongsourcemap)

# Steps to reproduce
1. run `sbt clean fastOptJS` first time
2. check generated source map file `./target/scala-2.11/scalajs-wrongsourcemap-fastopt.js.map`, there is such path to source file:
   ```
   "../src/main/scala/ru/lmars/bundler/Application.scala"
   ```
   this is wrong path!
3. run `sbt clean fastOptJS` second time
4. path to source file will be different:
   ```
   "../../src/main/scala/ru/lmars/bundler/Application.scala"
   ```
   and this is right path!
5. delete directory `./target/scala-2.11`
6. run `sbt clean fastOptJS`
7. source file path in .map file is broken again!

I ran into this bug many times when my file manager holds some directories inside `target` and `sbt clean` couldn't delete them (but successfully deletes `./target/scala-2.11`).

**Comment by sjrd** [comment]
Thanks for the report. I can indeed reproduce this. The reliable way to reproduce it is to start sbt (or `reload` it) at a time when the `target/scala-2.11` directory does not exist. This happens because of this line: https://github.com/scala-js/scala-js/blob/72456873a56f1ea833be996bb57707f2faecf52a/sbt-plugin/src/main/scala/org/scalajs/sbtplugin/ScalaJSPluginInternal.scala#L211
It is incorrect because `getParentFile` will return a path without a trailing `/`. Then, `.toURI` fixes it with a trailing `/` *but only if* the file already exists and is a directory. Since this line of code is evaluated at the time sbt is loaded, it will do the wrong thing if `target/scala-2.11` does not exist at that time.

The fix is to ensure that there is a trailing `/` in the URI in all cases.

**Comment by sjrd** [comment]
In fact, there is an even better way: don't use `getParentFile` at all. Relativizing URIs is perfectly capable of relativizing against a file rather than a directory. This is even more correct, since the browser will then resolve the relative URI based on the file URI, not its parent directory.

**Comment by sjrd** [comment]
@glmars Workaround that you can use right now:
```scala
scalaJSLinkerConfig := {
  val fastOptJSURI = (artifactPath in (Compile, fastOptJS)).value.toURI
  scalaJSLinkerConfig.value.withRelativizeSourceMapBase(fastOptJSURI)
}
```
In fact, you'll have to do that in 1.x anyway, because `relativeSourceMaps` does not exist there, replaced by `scalaJSLinkerConfig`.

**Comment by sjrd** [comment]
Fixed in 4d8f39a3c7dd304a07cada51bd908f57d99eafbb.

**Comment by glmars** [comment]
@sjrd, thank you for your attention even to such a minor (for someone) bug.
